code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplateBroadcastReceiver.java

```diff
@@ -0,0 +1,302 @@
+                    cachedImages.add(BitmapFactory.decodeStream(cachedCenterImage.getData()));
+                }
+            }
+
+            if (!StringUtils.isNullOrEmpty(rightImageUri)) {
+                final CacheResult cachedRightImage =
+                        cacheService.get(assetCacheLocation, rightImageUri);
+                if (cachedRightImage != null) {
+                    cachedImages.add(BitmapFactory.decodeStream(cachedRightImage.getData()));
+                }
+            }
+        }
+
+        final RemoteViews smallLayout =
+                new RemoteViews(packageName, R.layout.push_template_collapsed);
+        final RemoteViews expandedLayout =
+                new RemoteViews(packageName, R.layout.push_template_filmstrip_carousel);
+        smallLayout.setTextViewText(R.id.notification_title, titleText);
+        smallLayout.setTextViewText(R.id.notification_body, smallBodyText);
+        expandedLayout.setTextViewText(R.id.notification_title, titleText);
+        expandedLayout.setTextViewText(R.id.notification_body_expanded, expandedBodyText);
+
+        final String action = intent.getAction();
+        String newLeftCaption;
+        String newCenterCaption;
+        String newRightCaption;
+        Bitmap newLeftImage;
+        Bitmap newCenterImage;
+        Bitmap newRightImage;
+        String newLeftImageUri;
+        String newCenterImageUri;
+        String newRightImageUri;
+
+        if (action.equals(CampaignPushConstants.IntentActions.FILMSTRIP_LEFT_CLICKED)) {
+            newCenterImage = cachedImages.get(0);
+            newRightImage = cachedImages.get(1);
+            newLeftImage = cachedImages.get(2);
+            newCenterImageUri = leftImageUri;
+            newRightImageUri = centerImageUri;
+            newLeftImageUri = rightImageUri;
+            newLeftCaption = captionMap.get(rightImageUri);
+            newCenterCaption = captionMap.get(leftImageUri);
+            newRightCaption = captionMap.get(centerImageUri);
+        } else {
+            newCenterImage = cachedImages.get(2);
+            newRightImage = cachedImages.get(0);
+            newLeftImage = cachedImages.get(1);
+            newCenterImageUri = rightImageUri;
+            newRightImageUri = leftImageUri;
+            newLeftImageUri = centerImageUri;
+            newLeftCaption = captionMap.get(centerImageUri);
+            newCenterCaption = captionMap.get(rightImageUri);
+            newRightCaption = captionMap.get(leftImageUri);
+        }
+
+        expandedLayout.setImageViewBitmap(R.id.manual_carousel_filmstrip_center, newCenterImage);
+        expandedLayout.setImageViewBitmap(R.id.manual_carousel_filmstrip_left, newLeftImage);
+        expandedLayout.setImageViewBitmap(R.id.manual_carousel_filmstrip_right, newRightImage);
+        expandedLayout.setTextViewText(R.id.manual_carousel_filmstrip_caption, newCenterCaption);
+
+        // get custom color from hex string and set it the notification background
```

Comment:
Let us not allow methods to be very long with multiple responsibilities. Can you move the color setting logic into a separate private method. Can you also try to see other places where we can split methods that are super large into smaller ones.